python/lsst/dax/obscore/cli/cmd/commands.py

```diff
@@ -0,0 +1,57 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any
+
+import click
+from lsst.daf.butler.cli.opt import destination_argument, options_file_option, repo_argument
+from lsst.daf.butler.cli.utils import ButlerCommand, MWPath
+
+from ... import script
+
+
+@click.command(
+    short_help="Export Butler datasets as ObsCore Data Model",
+    cls=ButlerCommand,
+)
+@repo_argument(required=True)
+@destination_argument(
+    required=True,
+    help="DESTINATION is the location of the output file.",
+    type=MWPath(file_okay=True, dir_okay=False, writable=True),
+)
+@click.option(
+    "--config",
+    "-c",
+    type=MWPath(file_okay=True, dir_okay=False, exists=True),
+    help="Location of the configuration file in YAML format.",
+    required=True,
+)
+@click.option(
+    "--format",
+    help="Output format, one of 'parquet' or 'csv'.",
+    type=click.Choice(["csv", "parquet"]),
+    default="parquet",
+)
+@options_file_option()
+def obscore_export(*args: Any, **kwargs: Any) -> None:
```

Comment:
I was expecting the ability for users to say `--collections` `--datasets` and `--where` as for `butler query-datasets` so they can easily run test exports with small constrained subsets.

Reply:
Dataset types need to come from a config file as they have a bunch of options associated with them. Collections come from the same config file, and I can add command line option to override that config. "where" could be useful too, I'll add it as well, to both config and CLI.

Reply:
I agree that if you ask for a datasetType on the command line that is not present in the file then it should complain.